src/Symfony/Component/AssetMapper/Factory/MappedAssetFactory.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\Factory;
+
+use Symfony\Component\AssetMapper\AssetMapperCompiler;
+use Symfony\Component\AssetMapper\MappedAsset;
+use Symfony\Component\AssetMapper\Path\PublicAssetsPathResolverInterface;
+
+/**
+ * Creates MappedAsset objects by reading their contents & passing it through compilers.
+ */
+class MappedAssetFactory implements MappedAssetFactoryInterface
+{
+    private const PREDIGESTED_REGEX = '/-([0-9a-zA-Z]{7,128}\.digested)/';
+
+    private array $assetsCache = [];
+    private array $assetsBeingCreated = [];
+    private array $fileContentsCache = [];
+
+    public function __construct(
+        private PublicAssetsPathResolverInterface $assetsPathResolver,
+        private AssetMapperCompiler $compiler,
+    ) {
+    }
+
+    public function createMappedAsset(string $logicalPath, string $sourcePath): ?MappedAsset
+    {
+        if (\in_array($logicalPath, $this->assetsBeingCreated, true)) {
+            throw new \RuntimeException(sprintf('Circular reference detected while creating asset for "%s": "%s".', $logicalPath, implode(' -> ', $this->assetsBeingCreated).' -> '.$logicalPath));
+        }
+
+        if (!isset($this->assetsCache[$logicalPath])) {
+            $this->assetsBeingCreated[] = $logicalPath;
+
+            $asset = new MappedAsset($logicalPath);
+            $this->assetsCache[$logicalPath] = $asset;
+            $asset->setSourcePath($sourcePath);
+
+            $asset->setPublicPathWithoutDigest($this->assetsPathResolver->resolvePublicPath($logicalPath));
+            $publicPath = $this->getPublicPath($asset);
+            $asset->setPublicPath($publicPath);
+            [$digest, $isPredigested] = $this->getDigest($asset);
+            $asset->setDigest($digest, $isPredigested);
+            $asset->setContent($this->calculateContent($asset));
+
+            array_pop($this->assetsBeingCreated);
+        }
+
+        return $this->assetsCache[$logicalPath];
+    }
+
+    /**
+     * Returns an array of "string digest" and "bool predigested".
+     *
+     * @return array{0: string, 1: bool}
+     */
+    private function getDigest(MappedAsset $asset): array
+    {
+        // check for a pre-digested file
+        if (1 === preg_match(self::PREDIGESTED_REGEX, $asset->getLogicalPath(), $matches)) {
```

Comment:
```suggestion
        if (self::PREDIGESTED_REGEX, $asset->getLogicalPath(), $matches)) {
```

Reply:
the suggestion removes too many things

Reply:
you're correct - but I got it right when I applied it:

```php
if (preg_match(self::PREDIGESTED_REGEX, $asset->getLogicalPath(), $matches))
```